boards.txt

```diff
@@ -25,26 +25,26 @@ generic.serial.disableRTS=true
 
 generic.build.mcu=esp8266
 generic.build.f_cpu=80000000L
-generic.build.f_crystal=1
 generic.build.board=ESP8266_ESP01
 generic.build.core=esp8266
 generic.build.variant=generic
 generic.build.flash_mode=qio
 generic.build.spiffs_pagesize=256
 generic.build.debug_port=
 generic.build.debug_level=
+generic.build.extra_flags=-DF_CRYSTAL={build.f_crystal}
```

Comment:
probably... which version are you using?

If you are agree I proceed in this way:
- change #if statement in core_esp8266_phy.c as you suggest the first time
```c
    #if F_CRYSTAL == 40000000
      [48] = 0,
    #else
      [48] = 1,
    #endif
```
- remove *Crystal Freq.* menu from `boards.txt`
- update 'arduino-esp8266' entries to prevent Travis build failure